TestFoundation/TestProcess.swift

```diff
@@ -155,6 +154,13 @@ class TestProcess : XCTestCase {
         // Clear the environment to stop the malloc debug flags used in Xcode debug being
         // set in the subprocess.
         process.environment = [:]
+#if os(Android)
+        // In Android, we have to provide at least an LD_LIBRARY_PATH, or
+        // xdgTestHelper will not be able to find the Swift libraries.
+        if let ldLibraryPath = ProcessInfo.processInfo.environment["LD_LIBRARY_PATH"] {
+            process.environment?["LD_LIBRARY_PATH"] = ldLibraryPath
+        }
+#endif
         process.launch()
```

Comment:
Is the `LD_LIBRARY_PATH` still needed? When I updated `xdgTestHelper` to contain all of the utilities I updated the `CMakeLists.txt` to add an `rpath` to the build directory which should apply to both Linux and Android:
`set(XDG_TEST_HELPER_RPATH -Xlinker;-rpath;-Xlinker;${CMAKE_CURRENT_BINARY_DIR})`

This was needed because one of the tests clears the environment before calling `xdgTestHelper --env` and before adding in the `rpath` it was failing as the `LD_LIBRARY_PATH` was not set anymore.

Reply:
`${CMAKE_CURRENT_BINARY_DIR}` is a path in the build machine, but it doesn’t mean anything when moving the executable to the device for testing. That's also why this pieces are restricted to Android (they should not be needed for other targets).

Foundation CMakeLists is not prepared to test in Android, so the testing has to done manually, and while I always use the same directory, I would prefer not to hardcode anything in the CMakeLists for Android (I have to figure out how to simplify testing for Android).